### DISCUSSION TITLE
Scoop - Could not create process with command

### DISCUSSION THREAD

**Discussion by vunhatchuong** [JanDeDobbeleer/oh-my-posh#2989] [Troubleshoot] [answered]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

```bash
Could not create process with command '"C:\Users\ronny\scoop\apps\oh-my-posh\current\oh-my-posh.exe"  init pwsh --config C:\Users\ronny\.config\powershell\.\themes\catppuccin.omp.json'.
```

![image](https://user-images.githubusercontent.com/20581504/196845511-bb7822db-3002-4603-8877-8543f68c681c.png)

### Theme

catppuccin.omp.json

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
PS>oh-my-posh debug
Could not create process with command '"C:\Users\ronny\scoop\apps\oh-my-posh\current\oh-my-posh.exe"  debug'.
```

**Comment by JanDeDobbeleer** [comment]
@vunhatchuong123 did you restart your terminal? It now uses the actual installer and it should not even have that shim. How did you install oh-my-posh with scoop?

**Comment by vunhatchuong** [comment]
Yes I did restart my terminal, same error popup as when I reload profile on startup.

I install with 
```
scoop install https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/oh-my-posh.json
```

**Comment by JanDeDobbeleer** [comment]
@vunhatchuong123 this makes no sense, there's no shim to create here tbh. When you run `Get-Command oh-my-posh`, what doest it report?

**Comment by vunhatchuong** [comment]
@JanDeDobbeleer 
![image](https://user-images.githubusercontent.com/20581504/196866782-a7b39380-be4c-4f60-9d50-86b3e14f7b87.png)

**Comment by vunhatchuong** [comment]
Ok, idk what's going on but oh the original machine with the bug, I'm able to fix it by updating to 12.6.4

But on another machine with the same dotfiles, it's confusing.

<details>
<summary>omp installation process on another machine with same dotfiles</summary>

```bash
C:\Users\Ronny> scoop install https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/oh-my-posh.json
Installing 'oh-my-posh' (12.6.4) [64bit]
install-amd64.exe (6.7 MB) [==========================================================================================================================] 100%
Checking hash of install-amd64.exe ... ok.
Extracting install-amd64.exe ... done.
Running installer... error.
ERROR Exception calling "Start" with "0" argument(s): "An error occurred trying to start process 'C:\Users\Ronny\scoop\apps\oh-my-posh\12.6.4\install-amd64.exe' with working directory 'C:\Users\Ronny'. The system cannot find the file specified."
Installation aborted. You might need to run 'scoop uninstall oh-my-posh' before trying again.
C:\Users\Ronny> scoop install https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/oh-my-posh.json
WARN  Purging previous failed installation of oh-my-posh.
ERROR 'oh-my-posh' isn't installed correctly.
Removing older version (12.6.4).
'oh-my-posh' was uninstalled.
Installing 'oh-my-posh' (12.6.4) [64bit]
Loading install-amd64.exe from cache
Checking hash of install-amd64.exe ... ok.
Extracting install-amd64.exe ... done.
Running installer... error.
ERROR Exception calling "Start" with "0" argument(s): "An error occurred trying to start process 'C:\Users\Ronny\scoop\apps\oh-my-posh\12.6.4\install-amd64.exe' with working directory 'C:\Users\Ronny'. The system cannot find the file specified."
Installation aborted. You might need to run 'scoop uninstall oh-my-posh' before trying again.
C:\Users\Ronny> scoop install oh-my-posh
WARN  Purging previous failed installation of oh-my-posh.
ERROR 'oh-my-posh' isn't installed correctly.
Removing older version (12.6.4).
'oh-my-posh' was uninstalled.
Installing 'oh-my-posh' (12.6.3) [64bit] from main bucket
themes.zip (126.2 KB) [===============================================================================================================================] 100%
Checking hash of themes.zip ... ok.
posh-windows-amd64.exe (19.7 MB) [====================================================================================================================] 100%
Checking hash of posh-windows-amd64.exe ... ok.
Extracting themes.zip ... done.
Linking ~\scoop\apps\oh-my-posh\current => ~\scoop\apps\oh-my-posh\12.6.3
Creating shim for 'oh-my-posh'.
'oh-my-posh' (12.6.3) was installed successfully!
Notes
-----
Refer to 'https://ohmyposh.dev/docs/installation/prompt' for shell specific configurations.
C:\Users\Ronny> . $PROFILE
Remove-Alias: C:\Users\Ronny\.config\powershell\user_profile.ps1:54
Line |
  54 |  Remove-Alias ls
     |  ~~~~~~~~~~~~~~~
     | This command cannot find a matching alias because an alias with the name 'ls' does not exist.

 Ronny ~ scoop update *
Latest versions for all apps are installed! For more information try 'scoop status'
 Ronny ~ scoop update oh-my-posh
oh-my-posh: 12.6.3 (latest version)
Latest versions for all apps are installed! For more information try 'scoop status'
 Ronny ~ scoop install oh-my-posh@12.6.4
WARN  Given version (12.6.4) does not match manifest (12.6.3)
WARN  Attempting to generate manifest for 'oh-my-posh' (12.6.4)
Autoupdating oh-my-posh
Searching hash for themes.zip in https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v12.6.4/themes.zip.sha256
Found: f5568f7f3d35f0274c7268a40fb073312297f0615d265cabeb40653a59f5f026 using Extract Mode
Searching hash for posh-windows-386.exe in https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v12.6.4/posh-windows-386.exe.sha256
Found: 3141f47b8282f7c26e68b3a4ccd7c73d43587991e7493587d3bf62e7f599c14c using Extract Mode
Searching hash for themes.zip in https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v12.6.4/themes.zip.sha256
Found: f5568f7f3d35f0274c7268a40fb073312297f0615d265cabeb40653a59f5f026 using Extract Mode
Searching hash for posh-windows-amd64.exe in https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v12.6.4/posh-windows-amd64.exe.sha256
Found: b443500c9310f90f54ec86b675702e804269ce69775e1299f824c06cdb8dd01b using Extract Mode
Searching hash for themes.zip in https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v12.6.4/themes.zip.sha256
Found: f5568f7f3d35f0274c7268a40fb073312297f0615d265cabeb40653a59f5f026 using Extract Mode
Searching hash for posh-windows-arm64.exe in https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v12.6.4/posh-windows-arm64.exe.sha256
Found: 18485c7009c6fbfa5438788135f61f4b65f8cdb5af51268880a3428d1d5533dd using Extract Mode
Writing updated oh-my-posh manifest
Installing 'oh-my-posh' (12.6.4) [64bit]
themes.zip (126.2 KB) [===============================================================================================================================] 100%
Checking hash of themes.zip ... ok.
posh-windows-amd64.exe (19.7 MB) [====================================================================================================================] 100%
Checking hash of posh-windows-amd64.exe ... ok.
Extracting themes.zip ... done.
Linking ~\scoop\apps\oh-my-posh\current => ~\scoop\apps\oh-my-posh\12.6.4
Creating shim for 'oh-my-posh'.
'oh-my-posh' (12.6.4) was installed successfully!
```

</details>

Note: Before the first `scoop install` I already clean cache and everything.

TL;DR: Installing unsuccessful with recommended way.

```bash
scoop install https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/oh-my-posh.json
```
 I'm able to do it with `scoop install oh-my-posh@12.6.4` **specifically**

**Comment by JanDeDobbeleer** [comment]
@vunhatchuong123 I can also reproduce the last error. I specifically added that our installer was inno but then things go sideways. I'll have a look at other inno setup installers in the bucket and do adjustments needed. The problem with `scoop install oh-my-posh@12.6.4` is that it install an unsigned binary.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@vunhatchuong123 when you run the following command, does it return the same? Because it should. When using 12.6.3 (as 4 is indeed broken).

```
PS> scoop which oh-my-posh
~\AppData\Local\Programs\oh-my-posh\bin\oh-my-posh.exe
````

Try this, as it needs to remove that shim first:

- uninstall oh-my-posh: `scoop uninstall oh-my-posh` (this should remove the broken shim)
- reinstall the latest release: `scoop install https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/oh-my-posh.json`
- restart the terminal
- `scoop which oh-my-posh` should now point to `~\AppData\Local\Programs\oh-my-posh\bin\oh-my-posh.exe`

**Comment by vunhatchuong** [comment]
Yes both computer works with v12.6.3 now. And the output is at `~\AppData\Local\Programs\oh-my-posh\bin\oh-my-posh.exe`

**Comment by kamenminkov** [comment]
This happens to me as well from time to time (and I use Scoop too). It's like this now with `12.6.5`, but has happened before too. What I usually do is `scoop uninstall oh-my-posh && scoop install oh-my-posh@12.6.3` where `12.6.3` is a known working version before the one that causes problems. Then, after I see a new version, I do `scoop uninstall oh-my-posh && scoop install oh-my-posh` (as the instance that was installed using a specific version doesn't respond to `scoop update`).

**Reply by JanDeDobbeleer** [reply]
Be aware that that command installs non-signed binaries as the main bucket downloads the raw ones (and I will be removing those from the artifacts).
